File: NDOTestApp/NDOTravelExpenses-Step 6 VB/BusinessClasses/ExpenseVoucher.vb
Imports NDO
Imports System


<NDOPersistent()> _
Public Class ExpenseVoucher
    Inherits Expense

    ' Properties
    Public Overrides ReadOnly Property Amount() As Decimal
        Get
            Return _sum
        End Get
    End Property

    Public Property Sum() As Decimal
        Get
            Return _sum
        End Get
        Set(ByVal value As Decimal)
            _sum = value
        End Set
    End Property

    Public Overrides ReadOnly Property [Text]() As String
        Get
            Return _text
        End Get
    End Property

    Public Property VoucherText() As String
        Get
            Return _text
        End Get
        Set(ByVal value As String)
            _text = value
        End Set
    End Property


    ' Fields
    Private _sum As Decimal
    Private _text As String
End Class

File: Tutorial/NDOTestApp/NDOTravelExpenses-Step 6/BusinessClasses/ExpenseVoucher.cs
using System;
using NDO;
using System.ComponentModel;

namespace BusinessClasses
{
    [NDOPersistent, DisplayName("Expense Voucher")]
    public partial class ExpenseVoucher : Expense
    {
        string text;
        public string VoucherText
        {
            get { return text; }
            set { text = value; }
        }
        decimal sum;
        public decimal Sum
        {
            get { return sum; }
            set { sum = value; }
        }
        public ExpenseVoucher()
        {
        }
        public override decimal Amount
        {
            get { return sum; }
        }

        public override string Text
        {
            get { return text; }
        }
    }
}
